lib/resources/ssl.rb

```diff
@@ -44,6 +46,11 @@ class SSL < Inspec.resource(1)
   def initialize(opts = {})
     @host = opts[:host] ||
             inspec.backend.instance_variable_get(:@hostname)
+    # FIXME: This can be removed when/if @hostname is available as a property for 'Train::Transports::WinRM::Connection'
+    # Train enhancement request for this here: https://github.com/chef/train/issues/128
+    if @host.nil? && inspec.backend.class.to_s == 'Train::Transports::WinRM::Connection'
```

Comment:
It feels like we do not want to add this here, but instead we want to improve train directly. Could you push a PR for train, so that we can depend on that?